CADETProcess/transform.py

```diff
@@ -193,20 +193,28 @@ def untransform(self, x):
         ----------
         x : {float, array}
             Output parameter values.
+        precision : int, optional
+            Number of significant figures to which variable can be rounded.
+            If None, variable is not rounded. The default is None.
 
         Returns
         -------
         {float, array}
             Transformed parameter values.
         """
+        x_ = float(np.format_float_positional(x, precision=precision, fractional=False))
+
         if (
                 not self.allow_extended_output and
-                not np.all((self.lb <= x) * (x <= self.ub))):
+                not np.all((self.lb <= x_) * (x_ <= self.ub))):
             raise ValueError("Value exceeds output bounds.")
-        x = self._untransform(x)
+
+        x_ = self._untransform(x_)
+        x_ = float(np.format_float_positional(x_, precision=precision, fractional=False))
```

Comment:
Why is `x_` rounded twice? This is not entirely clear to me. I'd rather round it once at the end.